### PR TITLE
Is incident part of campaign

### PR DESCRIPTION
**Pull request by iyeshaya**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
https://github.com/demisto/etc/issues/36991

## Description
added the requested script 

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        Incident_type(str): the incident type name.
+
+    Returns:
+        List of ids as strings.
+    """
+    incidents_list = demisto.executeCommand("getIncidents", {'type': incident_type})[0]['Contents']['data']
+    if not incidents_list:
+        return []
+    # for type name with more than one word getIncidents returns also incidents with the type name of that is part of
+    # the original type name (Ex. Phishing Campaign will also return incidents of Phishing)
```
Comment (roysagi) [inline Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py:23]: good!

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+    Returns:
+        Campaign incident's id where at least one id from the incidents_ids_set is linked to that campaign incident.
+    """
+    for campaign_id in campaigns_ids_list:
+        campaign_context = demisto.executeCommand("getContext", {'id': campaign_id})[0]['Contents']
+        if isinstance(campaign_context, str):  # item not found
+            continue
+        campaign_context = campaign_context['context']
+        incidents_list = demisto.get(campaign_context, 'EmailCampaign.incidents')
```
Comment (roysagi) [inline Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py:61]: can we please verify with @idovandijk that all campaign incidents will have this context path to connect to other incidents?

Comment (roysagi) [inline Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py:61]:
```suggestion
        connected_incidents_list = demisto.get(campaign_context, 'EmailCampaign.incidents')
```

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        if incidents_list:
+            campaign_incidents_ids = {incident.get('id') for incident in incidents_list}
```
Comment (roysagi) [inline Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py:63]:
```suggestion
            connected_campaign_incidents_ids = {incident.get('id') for incident in incidents_list}
```

**Review by roysagi — APPROVED** [review]
Do look at my comments about variable naming and a few checks.
Other than that - good job!

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        campaigns_ids_list(List[str]): List of campaign incident ids to search in.
+        incidents_ids_set(Set[str]): Set of incident ids to search for.
+
+    Returns:
+        Campaign incident's id where at least one id from the incidents_ids_set is linked to that campaign incident.
+    """
+    for campaign_id in campaigns_ids_list:
+        campaign_context = demisto.executeCommand("getContext", {'id': campaign_id})[0]['Contents']
+        if isinstance(campaign_context, str):  # item not found
+            continue
```
[the remark this answers is not in the document]
Comment (iyeshaya) [reply Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py:59]: good point changed to try catch block. looks better and wont fail if the campaign_context has a different type

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py (hunk 2 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/IsIncidentPartOfCampaign/IsIncidentPartOfCampaign.py]: The incidents_list list is saved in the context so only campaign incidents with context are interesting to us.

**Review by roysagi — APPROVED** [review]
Looks good - please add a quick readme

File: Packs/Campaign/Scripts/IsIncidentPartOfCampaign/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+Get the incident campaign's ID for the campaign that is linked to at least one  of the given incidents.
+
+## Script Data
+---
+
+| **Name** | **Description** |
+| --- | --- |
+| Script Type | python3 |
+| Tags | phishing, campaign |
+| Demisto Version | 5.5.0 |
```
Comment (roysagi) [inline Packs/Campaign/Scripts/IsIncidentPartOfCampaign/README.md:10]:
```suggestion
| Cortex XSOAR Version | 5.5.0 |
```

**Review by roysagi — APPROVED** [review]

**Pull request merged**